ush/forecast_postdet.sh

```diff
@@ -0,0 +1,959 @@
+  $NCP -pf $ICSDIR/$CDATE/ocn/MOM*nc $DATA/INPUT/
+  
+  # Copy MOM6 fixed files
+  $NCP -pf $FIXmom/$OCNRES/* $DATA/INPUT/
+  
+  # Copy coupled grid_spec
+  $NCP -pf $FIX_DIR/fix_cpl/a${CASE}o${OCNRES}/grid_spec.nc $DATA/INPUT/
+  
+  # Copy mediator restart files to RUNDIR
+  if [ $inistep = 'restart' ]; then
+    $NCP $ROTDIR/$CDUMP.$PDY/$cyc/ufs.cpld*.nc $DATA/
+    $NCP $ROTDIR/$CDUMP.$PDY/$cyc/rpointer.cpl $DATA/
+  fi
+  
+  echo "SUB ${FUNCNAME[0]}: MOM6 input data linked/copied"
+}
+
+MOM6_nml() {
+  echo "SUB ${FUNCNAME[0]}: Creating name list for MOM6"
+  source $SCRIPTDIR/parsing_namelists_MOM6.sh
+  MOM6_namelists
+}
+
+MOM6_out() {
+  echo "SUB ${FUNCNAME[0]}: Copying output data for MOM6"
+  
+  export ENSMEM=${ENSMEM:-01}
+  
+  export IDATE=$CDATE
+  
+  if [ $RUN_ENVIR = "nco" ]; then
+    export COMIN=${COMIN:-$ROTDIR/$RUN.$PDY/$cyc}
+    export COMOUT=${COMOUT:-$ROTDIR/$RUN.$PDY/$cyc}
+  else
+    export COMIN="$ROTDIR/$CDUMP.$PDY/$cyc"
+    export COMOUT="$ROTDIR/$CDUMP.$PDY/$cyc"
+  fi
+  [[ ! -d $COMOUT ]] && mkdir -m 775 -p $COMOUT
+  
+  if [ $inistep = 'cold' ]; then
+    cp $DATA/ufs.cpld.cold.cpl.r.*.nc $COMOUT/
+    cp $DATA/rpointer.cpl $COMOUT/
+    status=$?
+    exit $status
+  else
+    if [ $FHRGRP -eq 0 ]; then
+      fhrlst="anl"
+    else
+      fhrlst=$(echo $FHRLST | sed -e 's/_/ /g; s/\[/ /g; s/\]/ /g; s/f/ /g; s/,/ /g')
+    fi
+  
+    # copy ocn files
+    for fhr in $fhrlst; do
+      export fhr=$fhr
+      if [[ 10#$fhr -ge 6 ]]; then
+        hh_inc_m=$((10#$FHOUT/2))
+        hh_inc_o=$((10#$FHOUT  ))
+  
+        # ------------------------------------------------------
+        #  adjust the dates on the mom filenames and save
+        # ------------------------------------------------------
+        VDATE=$($NDATE $fhr $IDATE)
+        YYYY=`echo $VDATE | cut -c1-4`
+        MM=`echo $VDATE | cut -c5-6`
+        DD=`echo $VDATE | cut -c7-8`
+        HH=`echo $VDATE | cut -c9-10`
+        SS=$((10#$HH*3600))
+        
+        m_date=$($NDATE -$hh_inc_m $VDATE)
+        p_date=$VDATE
+        
+        year=`echo $m_date | cut -c1-4`
+        month=`echo $m_date | cut -c5-6`
+        day=`echo $m_date | cut -c7-8`
+        hh=`echo $m_date | cut -c9-10`
+        
+        export ocnfile=ocn_${year}_${month}_${day}_${hh}.nc
+        
+        echo "$NCP -p $ocnfile $COMOUT/ocn$p_date.$ENSMEM.$IDATE.nc"
+        $NCP -p $ocnfile $COMOUT/ocn$p_date.$ENSMEM.$IDATE.nc
+        status=$?
+        [[ $status -ne 0 ]] && exit $status
+      fi
+    done
+    $NCP -p $DATA/ocn_daily*nc $COMOUT/
+    $NCP -p $DATA/wavocn*nc $COMOUT/ #temporary for p4
+    $NCP -p $DATA/INPUT/MOM_input $COMOUT/
+  fi
+}
+
+CICE_postdet() {
+  echo "SUB ${FUNCNAME[0]}: CICE after run type determination"
+  
+  year=$(echo $CDATE|cut -c 1-4)
+  month=$(echo $CDATE|cut -c 5-6)
+  day=$(echo $CDATE|cut -c 7-8)
+  stepsperhr=$((3600/$ICETIM))
+  nhours=$($NHOUR $CDATE ${year}010100)
+  istep0=$((nhours*stepsperhr))
```

Comment:
This istep0 line is only in this branch, not f/c-c, I think this was recently removed.

Reply:
Resolved.